crypto/evp/legacy_blake2.c

```diff
@@ -11,11 +11,31 @@
 #include "prov/blake2.h"        /* diverse BLAKE2 macros */
 #include "legacy_meth.h"
 
-#define ossl_blake2b_init ossl_blake2b512_init
-#define ossl_blake2s_init ossl_blake2s256_init
+/*
+ * Local hack to adapt the BLAKE2 init functions to what the
+ * legacy function signatures demand.
+ */
+static int blake2s_init(BLAKE2S_CTX *C)
+{
+    BLAKE2S_PARAM P;
+
+    ossl_blake2s_param_init(&P);
```

Comment:
This introduces an uninitialised read in `ossl_blake2s_param_init()`

Reply:
(Found by coverity)

Reply:
Argh...

Reply:
False positive...  `ossl_blake2s_param_init() intializes that very structure, it's utter silliness to initialize it before that!

Reply:
Actually not a false positive....but a mistake by me putting this comment on the wrong line. It's actually the call to `ossl_blake2b_param_init` below that is the uninit read.

Reply:
Did you perhaps mean `ossl_blake2b_init()` (i.e. no `param`)?

Reply:
Fix in #22103

Reply:
Oh, that hack.  Ok.  Well, I hope my fix is good enough